elastic/kibana
# [SharedUxChromeNavigation V2] Remove "v1" implementation

This PR replaces the "v1" version of the shared ux chrome navigation package with the new "v2" components introduced in https://github.com/elastic/kibana/pull/158297.

### Changes

* Replace old components with the new ones
* Added (temporarily) support for `href` along with `link` (deeplink id). This will be removed in a following PR
* Replaced the integration of the "v1" component in the `serverless_search` and `serverless_observability` side navigation.

### Notes for reviewer

* Apart from replacing the existing component and allow a `href` prop to be passed (and fixing the tests) there hasn't been an changed in the implementation  from https://github.com/elastic/kibana/pull/158297. Meaning that all the "changes" in the `navigation.stories.ts` files are simply the result of overriding version 1 with version 2.
* How to test:
  * Launch kibana for search and observability `yarn serverless oblt` | `yarn serverless es` and make sure the side navigation works as expected. 

Fixes https://github.com/elastic/kibana/issues/157707
